addons/procurement_jit/procurement_jit.py

```diff
@@ -1,34 +1,39 @@
 # -*- coding: utf-8 -*-
 # Part of Odoo. See LICENSE file for full copyright and licensing details.
 
+from openerp import api, models
 
-from openerp.osv import osv
 
-class procurement_order(osv.osv):
+class ProcurementOrder(models.Model):
     _inherit = "procurement.order"
 
-    def create(self, cr, uid, vals, context=None):
-        context = context or {}
-        procurement_id = super(procurement_order, self).create(cr, uid, vals, context=context)
-        if not context.get('procurement_autorun_defer'):
-            self.run(cr, uid, [procurement_id], context=context)
-            self.check(cr, uid, [procurement_id], context=context)
+    @api.model
+    def create(self, vals):
+        procurement_id = super(ProcurementOrder, self).create(vals)
+        if not self._context.get('procurement_autorun_defer'):
+            procurement_id.run()
+            procurement_id.check()
         return procurement_id
 
-    def run(self, cr, uid, ids, autocommit=False, context=None):
-        context = dict(context or {}, procurement_autorun_defer=True)
-        res = super(procurement_order, self).run(cr, uid, ids, autocommit=autocommit, context=context)
-
-        procurement_ids = self.search(cr, uid, [('move_dest_id.procurement_id', 'in', ids)], order='id', context=context)
-
+    @api.multi
+    def run(self, autocommit=False):
+        res = super(ProcurementOrder, self.with_context(procurement_autorun_defer=True)).run(autocommit=autocommit)
+        procurement_ids = self.with_context(procurement_autorun_defer=True).search([('move_dest_id.procurement_id', 'in', self.ids)], order='id')
```

Comment:
Variable name should be **procurements**  because now **search()** method returns recordset(s) instead of id(s) .